File: leetcode/C++/domino-and-tromino-tiling.cpp
class Solution {
public:
    int numTilings(int n) {
        vector<long> v = {0,1,2,5};
        for(int i=4; i<=n; ++i) v.push_back((2*v[i-1]+v[i-3])%1000000007);
        return v[n];
    }
};

File: leetcode/Python/domino-and-tromino-tiling.py
class Solution:
    def numTilings(self, n: int) -> int:
        sol = [0,1,2,5]
        for i in range(4, n+1):
            sol.append((sol[i-1]*2+sol[i-3])%1000000007)
        return sol[n]